app/src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java

```diff
@@ -0,0 +1,49 @@
+package pl.futurecollars.invoicing.db.memory;
+
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+import java.util.*;
+
+public class InMemoryDatabase implements Database {
+
+
+    public final Map<UUID, Invoice> database = new HashMap<>();
+
+    @Override
+    public Invoice save(Invoice invoice) {
+        Optional<Invoice> invoiceOptional = Optional.ofNullable(invoice);
+        if(invoiceOptional.isPresent()) {
+            while (database.containsKey(invoice.getId())) {
+                invoice.setId(UUID.randomUUID());
+            }
+            database.put(invoice.getId(), invoice);
+            return invoice;
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<Invoice> getById(UUID id) {
+        return Optional.ofNullable(database.get(id));
```

Comment:
It is good practice to use `Optional` when we have a method called such as `find...`. If we have method call `get...` we shouldn't expect `null` out of it